gslib/gcs_json_credentials.py

```diff
@@ -73,6 +77,75 @@
 GOOGLE_OAUTH2_DEFAULT_FILE_PASSWORD = 'notasecret'
 
 
+class PKCS12Signer(crypt_base.Signer, crypt_base.FromServiceAccountMixin):
+  """Signer for a p12 service account key."""
+
+  def __init__(self, key):
+    self._key = key
+
+  # Defined in crypt_base.Signer interface.
+  # It is not used by p12 service account keys.
+  @property
+  def key_id(self):
+    return None
+
+  def sign(self, message):
+    message = _helpers.to_bytes(message)
+    from google.auth.crypt import _cryptography_rsa  # pylint: disable=g-import-not-at-top
+    return self._key.sign(
+        message,
+        _cryptography_rsa._PADDING,  # pylint: disable=protected-access
+        _cryptography_rsa._SHA256)  # pylint: disable=protected-access
+
+  @classmethod
+  def from_string(cls, key_strings, key_id=None):
+    del key_id
```

Comment:
Typically we add a comment explaining why we are deleting it, e.g "del key_id  # Unused"

Reply:
Done